### PR TITLE
Add test reusable workflow

### PR DESCRIPTION
**Pull request by damacus**
# Description

Add test reusable workflow

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: .github/workflows/test-kitchen.yml [hunk 1]
```diff
@@ -1,40 +1,17 @@
...
-
-  mdl:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check out code
-        uses: actions/checkout@v2
-      - name: Run Markdown Lint
-        uses: actionshub/markdownlint@main
+  lint-unit:
+    uses: sous-chefs/.github/.github/workflows/lint-unit.yml@c4d27f78aaae5e927289b1d7f0c4094ba8cf2846
```
Comment (Xorima) [inline .github/workflows/test-kitchen.yml:11]: Not 100% sold on this location.

Comment (ramereth) [reply .github/workflows/test-kitchen.yml:11]: Why does it need to use a hash here instead of a branch name?

Comment (damacus) [reply .github/workflows/test-kitchen.yml:11]: Happy for it to be either. This was just showing that you could do either 👍🏼

Comment (pschaumburg) [reply .github/workflows/test-kitchen.yml:11]: Fun fact: I'd also place it in the exact same directory for our tecRacer cookbooks etc..
So my opinion: before creating a new repo only for those reusable workflows, which probably could become an option in the future to be stored in that directory, I'd place them there as well :-)

**Review by Xorima — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [10.1.2](https://github.com/sous-chefs/java/releases/tag/10.1.2)

**Pull request merged**